sdks/python/apache_beam/transforms/core.py

```diff
@@ -3962,9 +3963,46 @@ def to_runner_api_parameter(self, context):
   def infer_output_type(self, unused_input_type):
     if not self.values:
       return typehints.Any
-    return typehints.Union[[
-        trivial_inference.instance_to_type(v) for v in self.values
-    ]]
+
+    try:
+      if not all(isinstance(v, pvalue.Row) for v in self.values):
+        raise TypeError("All data must be Rows.")
```

Comment:
this introduces stricter schema validation for beam.Row elements, which could be a breaking change. I am wondering whether this is too aggressive. Shall we just focus on the null-able fields for now?

Reply:
Sounds reasonable, thanks